pkg/api/v1/pod/util_test.go

```diff
@@ -404,3 +405,140 @@ func TestIsPodAvailable(t *testing.T) {
 		}
 	}
 }
+
+func TestGetContainerStatus(t *testing.T) {
+	type ExpectedStruct struct {
+		status v1.ContainerStatus
+		exists bool
+	}
+
+	tests := []struct {
+		status   []v1.ContainerStatus
+		name     string
+		expected ExpectedStruct
+	}{
+		{
+			status:   []v1.ContainerStatus{{Name: "test1", Ready: false, Image: "image1"}},
+			name:     "test1",
+			expected: ExpectedStruct{status: v1.ContainerStatus{Name: "test1", Ready: false, Image: "image1"}, exists: true},
+		},
+		{
+			status:   []v1.ContainerStatus{{Name: "test2", Ready: false, Image: "image2"}},
+			name:     "test1",
+			expected: ExpectedStruct{status: v1.ContainerStatus{}, exists: false},
+		},
+		{
+			status:   []v1.ContainerStatus{{Name: "test3", Ready: false, Image: "image3"}},
+			name:     "",
+			expected: ExpectedStruct{status: v1.ContainerStatus{}, exists: false},
+		},
+		{
+			status:   nil,
+			name:     "",
+			expected: ExpectedStruct{status: v1.ContainerStatus{}, exists: false},
+		},
+	}
+
+	for _, test := range tests {
+		resultStatus, exists := GetContainerStatus(test.status, test.name)
+		assert.Equal(t, test.expected.status, resultStatus, "unexpected test: %v, expected status: %v, actual: %v", test, test.expected.status, resultStatus)
```

Comment:
No need to print the expected and resultStatus yourself. `assert` will do that if it fails.